BranchMetrics/AdobeBranchExtension-Android
# SDK-281 -- Adobe Launch updates

## Reference
SDK-281 -- Adobe Launch updates to match iOS

## Description
Issue 1:  Unpin specific Adobe SDK versions
Issue 2:  Add a method to "whitelist" events

## Testing Instructions
Sample continues to function.

The Sample has a new test feature, where if you long-press the share button it will try to send two events, one that should get dropped and one that should succeed.   You can view the logs and see the behavior.

## Risk Assessment [`MEDIUM`]
Unpinning SDK versions is low risk.  Whitelisting events is a breaking change for existing apps that expect their custom events to succeed without knowing that they now need those to be whitelisted.


## Reviewer Checklist (To be checked off by the reviewer only)

**You can only merge once all of these are checked off!**

- Correctness & Style
